src/traits.rs

```diff
@@ -136,6 +136,23 @@ where
 ///# Ok(())
 ///# }
 ///```
+///
+/// # Heap allocation
+///
+/// For systems where stack space is limited but heap space is available
+/// (i.e. ESP32 platforms with PSRAM) it's possible to allocate the display buffer
+/// in heap; the Displayxxxx:default() is implemented as always inline, so you can
+/// use Box::new to request heap space for the display buffer.
+///
+///```rust, no_run
+///let mut display = Box::new(Display4in2::default());
+///let _ = Line::new(Point::new(0, 120), Point::new(0, 295))
+///     .into_styled(PrimitiveStyle::with_stroke(Color::Black, 1))
+///     .draw(&mut *display);
+///
+///epd.update_frame(&mut spi, display.buffer(), &mut delay)?;
+///epd.display_frame(&mut spi, &mut delay)?;
```

Comment:
```suggestion
```

just remove these lines, since they bring no immediate benefit here

Reply:
If you use `cargo test` before commiting it should show you the necessary parts to fix as well.